docs/docs/30-how-to-guides/14-working-with-stages.md

```diff
@@ -0,0 +1,548 @@
+  - origin:
+      kind: Warehouse
+      name: microservice-a
+    sources:
+      direct: true
+  - origin:
+      kind: Warehouse
+      name: microservice-b
+    sources:
+      direct: true
+  # ...
+```
+
+:::tip
+By requesting `Freight` from multiple sources, a `Stage` can effectively
+participate in _multiple pipelines_ that may each deliver different collections
+of artifacts independently of the others.
+:::
+
+### Promotion Templates
+
+The `spec.promotionTemplate` field is used to describe _how_ to transition
+`Freight` into the `Stage`. The `spec.promotionTemplate.steps` field describes
+the discrete steps of a promotion process in detail.
+
+In the following, very common example, the `promotionTemplate` describes steps
+to:
+
+1. Clone a Git repository containing Kubernetes manifests and Kustomize
+   configuration, checking out two different branches to two different
+   directories.
+
+1. Clears the contents of one working tree, with intentions to fully replace its
+   contents.
+
+1. Runs the equivalent of `kustomize edit set image` to update a
+   `kustomization.yaml` file with a reference to an updated
+   `public.ecr.aws/nginx/nginx` container image.
+
+1. Renders the updated manifests using the equivalent of `kustomize build`.
+
+1. Commits the updated manifests and pushes them to the `stage/test` of the
+   remote repository.
+
+1. Forces Argo CD to sync the `kargo-demo-test` application to the latest commit
+   of the `stage/test` branch.
+
+```yaml
+promotionTemplate:
+  spec:
+    vars:
+    - name: gitopsRepo
+      value: https://github.com/example/repo.git
+    - name: imageRepo
+      value: public.ecr.aws/nginx/nginx
+    - name: srcPath
+      value: ./src
+    - name: outPath
+      value: ./out
+    - name: targetBranch
+      value: stage/${{ ctx.stage }}
+    steps:
+    - uses: git-clone
+      config:
+        repoURL: \${{ vars.gitopsRepo }}
+        checkout:
+        - branch: main
+          path: \${{ vars.srcPath }}
+        - branch: stage/\${{ ctx.stage }}
+          create: true
+          path: \${{ vars.outPath }}
+    - uses: git-clear
+      config:
+        path: \${{ vars.outPath }}
+    - uses: kustomize-set-image
+      as: update-image
+      config:
+        path: \${{ vars.srcPath }}/base
+        images:
+        - image: \${{ vars.imageRepo }}
+    - uses: kustomize-build
+      config:
+        path: \${{ vars.srcPath }}/stages/\${{ ctx.stage }}
+        outPath: \${{ vars.outPath }}/manifests.yaml
+    - uses: git-commit
+      as: commit
+      config:
+        path: \${{ vars.outPath }}
+        messageFromSteps:
+        - update-image
+    - uses: git-push
+      config:
+        path: \${{ vars.outPath }}
+        branch: \${{ vars.targetBranch }}
+    - uses: argocd-update
+      config:
+        apps:
+        - name: kargo-demo-\${{ ctx.stage }}
+          sources:
+          - repoURL: \${{ vars.gitopsRepo }}
+            desiredRevision: \${{ outputs.commit.commit }}
```

Comment:
The escaping of the `$` character isn't necessary here. If you've seen it in other places, they are places where there was YAML being pasted into a terminal and the shell had to be made to ignore the `$`.

```suggestion
        repoURL: ${{ vars.gitopsRepo }}
        checkout:
        - branch: main
          path: ${{ vars.srcPath }}
        - branch: stage/${{ ctx.stage }}
          create: true
          path: ${{ vars.outPath }}
    - uses: git-clear
      config:
        path: ${{ vars.outPath }}
    - uses: kustomize-set-image
      as: update-image
      config:
        path: ${{ vars.srcPath }}/base
        images:
        - image: ${{ vars.imageRepo }}
    - uses: kustomize-build
      config:
        path: ${{ vars.srcPath }}/stages/${{ ctx.stage }}
        outPath: ${{ vars.outPath }}/manifests.yaml
    - uses: git-commit
      as: commit
      config:
        path: ${{ vars.outPath }}
        messageFromSteps:
        - update-image
    - uses: git-push
      config:
        path: ${{ vars.outPath }}
        branch: ${{ vars.targetBranch }}
    - uses: argocd-update
      config:
        apps:
        - name: kargo-demo-${{ ctx.stage }}
          sources:
          - repoURL: ${{ vars.gitopsRepo }}
            desiredRevision: ${{ outputs.commit.commit }}
```